### PR TITLE
[operator] Introduce Gardener control plane configuration options to `Garden` API

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
This PR augments the `Garden` API with configuration options for the Gardener control plane components. For now, we only expose a very basic minimal set (similar to how we only expose the really necessary options for the Kubernetes control plane components). We can extend it further as needed in the future.

This PR only modifies the API and does not introduce any changes to the controllers.

**Which issue(s) this PR fixes**:
Part of #7016 

**Special notes for your reviewer**:
/cc @ScheererJ @timuthy @oliver-goetz 

**Release note**:

```breaking operator
When deploying this version of `gardener-operator`, make sure that you update your `Garden` resources with the new `.spec.virtualCluster.gardener.clusterIdentity` field. If you already have a `gardener-apiserver` deployment, make sure that the value matches the `--cluster-identity` flag of the current `gardener-apiserver` deployment.
```

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/types.go [hunk 1]
```diff
@@ -398,6 +401,75 @@ type KubeControllerManagerConfig struct {
...
+	// +optional
+	AuditConfig *gardencorev1beta1.AuditConfig `json:"auditConfig,omitempty"`
+	// Logging contains configuration for the log level and HTTP access logs.
+	// +optional
+	Logging *gardencorev1beta1.KubeAPIServerLogging `json:"logging,omitempty"`
+	// Requests contains configuration for request-specific settings for the kube-apiserver.
+	// +optional
+	Requests *gardencorev1beta1.KubeAPIServerRequests `json:"requests,omitempty"`
+	// WatchCacheSizes contains configuration of the API server's watch cache sizes.
+	// Configuring these flags might be useful for large-scale Shoot clusters with a lot of parallel update requests
```
Comment (ScheererJ) [inline pkg/apis/operator/v1alpha1/types.go]: Is this really about the shoot clusters, but not rather the seeds?

```suggestion
	// Configuring these flags might be useful for large-scale Garden clusters with a lot of parallel update requests
```

File: pkg/apiserver/plugins.go [hunk 2]
```diff
@@ -49,69 +50,34 @@ import (
...
+		plugin.PluginNameOpenIDConnectPreset,             // OpenIDConnectPreset
+		plugin.PluginNameClusterOpenIDConnectPreset,      // ClusterOpenIDConnectPreset
+		plugin.PluginNameCustomVerbAuthorizer,            // CustomVerbAuthorizer
+		plugin.PluginNameManagedSeed,                     // ManagedSeed
+		plugin.PluginNameManagedSeedShoot,                // ManagedSeedShoot
+		plugin.PluginNameBastion,                         // Bastion
+		mutatingwebhook.PluginName,                       // MutatingAdmissionWebhook
+		validatingwebhook.PluginName,                     // ValidatingAdmissionWebhook
+		validatingadmissionpolicy.PluginName,             // ValidatingAdmissionPolicy
+		resourcequota.PluginName,                         // ResourceQuota
```
Comment (ScheererJ) [inline pkg/apiserver/plugins.go]: Why did you move `AllOrderedPlugins` into the `plugin` module while `DefaultOnPlugins` stays here? Would it not be more consistent to have them both in one place as before?

File: pkg/apis/core/validation/shoot.go [hunk 3]
```diff
@@ -949,6 +949,35 @@ func ValidateKubeAPIServerLogging(loggingConfig *core.KubeAPIServerLogging, fldP
 	return allErrs
 }
 
+// ValidateKubeAPIServerRequests validates the given KubeAPIServer request fields.
+func ValidateKubeAPIServerRequests(requests *core.KubeAPIServerRequests, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if requests != nil {
+		const maxMaxNonMutatingRequestsInflight = 800
```
Comment (ScheererJ) [inline pkg/apis/core/validation/shoot.go]: Not trying to offend any Max out there, but I would still prefer to have only one Max here. How about `maxNonMutatingRequestsInflightLimit` instead.

```suggestion
		const maxNonMutatingRequestsInflightLimit = 800
```

File: pkg/apis/core/validation/shoot.go [hunk 4]
```diff
@@ -949,6 +949,35 @@ func ValidateKubeAPIServerLogging(loggingConfig *core.KubeAPIServerLogging, fldP
...
+		if v := requests.MaxNonMutatingInflight; v != nil {
+			path := fldPath.Child("maxNonMutatingInflight")
+
+			allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*v), path)...)
+			if *v > maxMaxNonMutatingRequestsInflight {
+				allErrs = append(allErrs, field.Invalid(path, *v, fmt.Sprintf("cannot set higher than %d", maxMaxNonMutatingRequestsInflight)))
+			}
+		}
+
+		const maxMaxMutatingRequestsInflight = 400
```
Comment (ScheererJ) [inline pkg/apis/core/validation/shoot.go]: Same as above

**Review by ScheererJ — COMMENTED** [review]
Thanks for the change. I have only a few minor comments.

File: pkg/apis/core/validation/shoot.go (hunk 3 above)
Comment (rfranzke) [reply pkg/apis/core/validation/shoot.go]: Technically, it's correct to have two `max` 😉 But I can change it, if you want.

Comment (ScheererJ) [reply pkg/apis/core/validation/shoot.go]: My Max stack is limited to one. Hence, I would appreciate if you prevent an overflow of my Max stack ;-) .

**Review by ScheererJ — COMMENTED** [review]
/lgtm

**Comment by rfranzke** [comment]
Can we go ahead with this PR?

**Comment by rfranzke** [comment]
/approve
As discussed in the daily
